src/globus_sdk/client.py

```diff
@@ -129,14 +212,21 @@ def get(  # pylint: disable=missing-param-doc
         *,
         query_params: dict[str, t.Any] | None = None,
         headers: dict[str, str] | None = None,
+        ignore_authorizer: bool = False,
```

Comment:
Hmm, I can get behind this approach (I assume this is for GCS client id lookup) but I don't love the name. It references an SDK specific concept of an `authorizer` where I think it could convey the same information to a broader audience if instead it used the more generic term `authorization` (referring to both the concept and the oauth2 header).

How do you feel about `skip_authorization`, `omit_authorization`, or even `include_authorization` if we wanted to invert the default to be `True`?

Reply:
Yeah this is for GCS client id lookup.

Part of the reason I went with this name was for a hypothetical use case of passing an explicit `Authorization` header in `headers` that you don't want overwritten by the `authorizer`. Calling it `skip_authorization` or `omit_authorization` might imply that such an `Authorization` header might still be omitted.

Maybe something like `automatic_authorization` with a default of `True`?

Reply:
Okay, I hadn't considered that possibility.

I can be happy with either ignore_authorizer as you initially had it or automatic_authorization in that case. Not much of a preference in either direction.

Reply:
I have an alternative to offer on this, which is to make it a capability of the `RequestsTransport` which can be accessed via `tune()`.
e.g., in GCSClient:
```python
with self.transport.tune(automatic_authorization=False):
    return self.get(...)
```

The advantage I see in this is that it avoids exposing an underlying detail of behavior in a very prominent position (the `BaseClient.get` method), and instead nests it in a way that makes it more of an "advanced" option.

Reply:
I don't love the contextmanager interface for this but preference aside, I also don't think it actually would workwork (at least in the current structure). 

Disabling `automatic_authorization` must cause GlobusApp to skip *creation* of an authorizer, not just the use of it. That's handled by the BaseClient, not the transport layer & pushing that into the transport layer would mean attaching both the resource server & globus app to that request interface.

Reply:
I put together a quick sample of how this would work here:
https://github.com/sirosen/globus-sdk-python/tree/allow-suppress-automatic-authz

Reply:
I like the idea of putting this in the Transport layer, but we would still need some way to skip trying to create the `authorizer` at the client level for this to be useful for `GCSClient` so I don't think we'd be saving much interface wise

Reply:
Alternate way uglier, but less impactful idea.

What if `get_gcs_info` just didn't use the standard `BaseClient.get` and  by the transitive property `BaseClient.request` methods?

We could either copy/paste the logic inside of request or shove it into a function which accepts a transport + request params & don't expose that publicly.

Reply:
Bear in mind though that we do have one more use case which requires this style of "retrieve client/scope info without any authorization" which is the to-be-built ActionProviderClient.

Reply:
```python

class BaseClient:
   ...
   def request(
        self,
        method: str,
        path: str,
        *,
        query_params: dict[str, t.Any] | None = None,
        data: _DataParamType = None,
        headers: dict[str, str] | None = None,
        encoding: str | None = None,
        allow_redirects: bool = True,
        stream: bool = False,
    ) -> GlobusHTTPResponse:
        return self._request(method, path, query_params=query_params, data=data, headers=headers, encoding=encoding, allow_redirects=allow_redirects, stream=stream)

    def _request(
        self,
        method: str,
        path: str,
        *,
        query_params: dict[str, t.Any] | None = None,
        data: _DataParamType = None,
        headers: dict[str, str] | None = None,
        encoding: str | None = None,
        allow_redirects: bool = True,
        stream: bool = False,
        automatic_authorization: bool = True,
    ) -> GlobusHTTPResponse:
        ... regular transport layer invocation ...
```

then `get_gcs_info` would just call the internal route

```python
class GCSClient:
  def get_gcs_info(self) -> UnpackingGCSResponse:
      return self._request(...)
```

Reply:
I'm personally not too bothered by the impact, anyone working at the `BaseClient.get` level is already a fairly advanced user, and I think it makes sense to have this available at the same interface for providing custom headers

Reply:
Kk, I'm happy with the way it's structured as well.

I'm clicking approve on this PR; but will leave this thread unresolved for Stephen to continue the discussion if he'd like to.

Reply:
With the rename to `automatic_authorization` I'm less bothered by this. I feel like `request()` and its siblings have a lot of options, but... is that bad? It seems fine, on a second read.

Thanks for indulging me in some thinking and speculation! Approving!